### PR TITLE
test: Add React 18 to test matrix

### PR DESCRIPTION
**Pull request by eps1lon**

### CODE REVIEW

File: test/TransitionGroup-test.js
Comment (eps1lon) [inline test/TransitionGroup-test.js:95]: The second `appear` is new i.e. there is an additional `onEnter` call.

File: test/TransitionGroup-test.js
Comment (eps1lon) [inline test/TransitionGroup-test.js:102]: The second `enter` is new i.e. there is an additional `onEnter` call.

**Comment by koba04** [comment]
@eps1lon 
React v18 has been released, so it's the time to test with React v18. Could you make this PR ready for review?

**Comment by eps1lon** [comment]
> @eps1lon React v18 has been released, so it's the time to test with React v18. Could you make this PR ready for review?

React 18 was not released yet. Only a release **candidate** is available. I've updated the branch but it shouldn't be merged yet considering it's only using a release candidate.

**Comment by koba04** [comment]
@eps1lon 
> React 18 was not released yet. 
Yes, it was a typo. 

>  I've updated the branch but it shouldn't be merged yet considering it's only using a release candidate.

Yes, we don't drop support for older versions of React, so we shouldn't update React in `package.json`.
At this time, we should run tests with React v16, v17, and v18 RC, so it would be nice if we could add React v18 in [the env](https://github.com/reactjs/react-transition-group/blob/master/.travis.yml#L5-L7) and pass all tests with all versions.
Is it possible?

**Comment by eps1lon** [comment]
> Is it possible?

Sure, but we're blocked by https://github.com/reactjs/react-transition-group/pull/771. I've tried contacting @jquense in the past (via GitHub ping and Twitter DM) but never got a response. Not sure how to reach him.

**Comment by koba04** [comment]
@eps1lon 
> Sure, but we're blocked by #771. I've tried contacting @jquense in the past (via GitHub ping and Twitter DM) but never got a response. Not sure how to reach him.

Yeah, I'll reach him again.
We can only release new versions by `semantic-release` on TravisCI, so we can't migrate TravisCI to GitHub Actions without his help.
What about adding React v18 RC as the target of TravisCI?

**Comment by jquense** [comment]
I have been summoned. Sorry if i've ignored anyone, it's inadvertent, i've just not have a ton of OSS time or energy lately.

What do y'all need from me? I'm happy to hand permissions or move stuff around as needed. Also from a maintenance standpoint, I'd just drop react 16 support at this point. it's been over a year since 17 was released, make y'alls life easier.

**Comment by eps1lon** [comment]
> I have been summoned. Sorry if i've ignored anyone, it's inadvertent, i've just not have a ton of OSS time or energy lately.

Hey,

https://github.com/reactjs/react-transition-group/pull/772 needs your attention. Specifically, the environment secrets from Travis CI need to be ported to GitHub actions. I think this mainly affects semantic-release i.e. a npm release token.

**Comment by jquense** [comment]
I've added @koba04 to the npm package, they should now be able to use their token for any CI setup 👍

**Comment by koba04** [comment]
@jquense 
Thank you! Could you add me as an admin of this GitHub repository?
We have a plan to migrate CI to GitHub Actions. That requires setting an npm token to the GitHub repository and it requires an admin privilege of the repository.

I also want to add @eps1lon as a maintainer of this repository.

**Comment by koba04** [comment]
@eps1lon I've invited you as an admin of the GitHub and npm of this package, so we can move this forward👍

File: test/TransitionGroup-test.js
Comment (eps1lon) [inline test/TransitionGroup-test.js:96]: See https://github.com/reactwg/react-18/discussions/19

File: test/CSSTransition-test.js
Comment (koba04) [inline test/CSSTransition-test.js:241]: Should we add `++count` and verify it in the later `waitFor` callback to confirm this `onEntered` was called? Otherwise, this test would be passed even if the `onEntered` hasn't been called.

File: test/CSSTransition-test.js
Comment (koba04) [inline test/CSSTransition-test.js:299]: Same as the above.

File: test/CSSTransition-test.js
Comment (koba04) [inline test/CSSTransition-test.js:340]: Same as the above.

File: test/CSSTransition-test.js
Comment (koba04) [inline test/CSSTransition-test.js:371]: Same as the above.

File: test/SwitchTransition-test.js
Comment (koba04) [inline test/SwitchTransition-test.js:111]: Why is this call added?

File: test/Transition-test.js
Comment (koba04) [inline test/Transition-test.js:338]: Should we have to verify this `onEntered` has been called?

File: test/TransitionGroup-test.js
Comment (koba04) [inline test/TransitionGroup-test.js:78]:
```suggestion
  it('should handle transitioning correctly', () => {
```

File: test/TransitionGroup-test.js
Comment (koba04) [reply test/TransitionGroup-test.js:96]: 👍  I hope React has a way to detect the StrictEffects mode. 😄

**Review by koba04 — COMMENTED** [review]
Thank you for your great works 👍 👍 
I've left some comments, so please confirm them🙏

File: test/CSSTransition-test.js
Comment (eps1lon) [reply test/CSSTransition-test.js:241]: These checks and the ones mentioned later all moved further down in the test e.g.
```ts
await waitFor(() => {
    expect(count).toEqual(2);
});
```

Does that make sense?

File: test/SwitchTransition-test.js
Comment (eps1lon) [reply test/SwitchTransition-test.js:111]: Let me check. I think it was due to previous tests relying on implementation details related to React scheduling timings. Will check what specifically was happening.

File: test/CSSTransition-test.js
Comment (koba04) [reply test/CSSTransition-test.js:241]: Yeah, that makes sense  for me 👍  Thank you!

**Comment by koba04** [comment]
I'll approve this when https://github.com/reactjs/react-transition-group/pull/771#discussion_r789766252 has been addressed.

File: test/Transition-test.js
Comment (eps1lon) [reply test/Transition-test.js:338]: This is similar to https://github.com/reactjs/react-transition-group/pull/771#discussion_r791570073:

This condition moved down to 

```js
     await waitFor(() => {
        expect(count).toEqual(2);
      });
```

**Comment by eps1lon** [comment]
> I'll approve this when [#771 (comment)](https://github.com/reactjs/react-transition-group/pull/771#discussion_r789766252) has been addressed.

See https://github.com/reactjs/react-transition-group/pull/771#discussion_r807317754

**Review by koba04 — APPROVED** [review]
Thank you, LGTM! 👍

**Pull request merged**
